amazon/ion/core.py

```diff
@@ -122,7 +122,7 @@ def derive_field_name(self, field_name):
         Args:
             field_name (Union[amazon.ion.symbols.SymbolToken, unicode]): The field name to set.
         Returns:
-            IonEvent: The newly generated event.
+            IonEvent: The newly generated non-thunk event.
```

Comment:
Actually if you look below the generated event is based on the event derived, so it could be a thunk event.